### PR TITLE
Fix research disk crash

### LINKED ISSUE #33188: Puting point disk in R&D server crashes game
## Description
Putting any point disk in R&D server crashes game.

**Reproduction**
Put any point disk in an R%D server

**Screenshots**
```
[FATL] unhandled: Robust.Shared.Utility.DebugAssertException: Exception of type 'Robust.Shared.Utility.DebugAssertException' was thrown.
   at Robust.Shared.Utility.DebugTools.Assert(Boolean condition) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\Utility\DebugTools.cs:line 37
   at Content.Shared.Interaction.SharedInteractionSystem.InteractDoAfter(EntityUid user, EntityUid used, Nullable`1 target, EntityCoordinates clickLocation, Boolean canReach, Boolean checkDeletion) in C:\Users\Sol\source\repos\space-station-14\Content.Shared\Interaction\SharedInteractionSystem.cs:line 1065
   at Content.Shared.Interaction.SharedInteractionSystem.InteractUsing(EntityUid user, EntityUid used, EntityUid target, EntityCoordinates clickLocation, Boolean checkCanInteract, Boolean checkCanUse) in C:\Users\Sol\source\repos\space-station-14\Content.Shared\Interaction\SharedInteractionSystem.cs:line 1025
   at Content.Shared.Interaction.SharedInteractionSystem.UserInteraction(EntityUid user, EntityCoordinates coordinates, Nullable`1 target, Boolean altInteract, Boolean checkCanInteract, Boolean checkAccess, Boolean checkCanUse) in C:\Users\Sol\source\repos\space-station-14\Content.Shared\Interaction\SharedInteractionSystem.cs:line 440
   at Content.Shared.Interaction.SharedInteractionSystem.HandleUseInteraction(ICommonSession session, EntityCoordinates coords, EntityUid uid) in C:\Users\Sol\source\repos\space-station-14\Content.Shared\Interaction\SharedInteractionSystem.cs:line 314
   at Robust.Shared.Input.Binding.PointerInputCmdHandler.<>c__DisplayClass5_0.<.ctor>b__0(PointerInputCmdArgs& args) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\Input\Binding\InputCmdHandler.cs:line 97
   at Robust.Shared.Input.Binding.PointerInputCmdHandler.HandleCmdMessage(IEntityManager entManager, ICommonSession session, IFullInputCmdMessage message) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\Input\Binding\InputCmdHandler.cs:line 127
   at Robust.Server.GameObjects.InputSystem.InputMessageHandler(InputCmdMessage message, EntitySessionEventArgs eventArgs) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\GameObjects\EntitySystems\InputSystem.cs:line 65
   at Robust.Shared.GameObjects.EventBusExt.HandlerWrapper`1.Invoke(EntitySessionMessage`1 msg) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EventBusExt.cs:line 46
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass4_0`1.<SubscribeEvent>b__0(Unit& ev) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EntityEventBus.Broadcast.cs:line 173
   at Robust.Shared.GameObjects.EntityEventBus.ProcessSingleEventCore(EventSource source, Unit& unitRef, EventData subs) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EntityEventBus.Broadcast.cs:line 335
   at Robust.Shared.GameObjects.EntityEventBus.ProcessSingleEvent(EventSource source, Unit& unitRef, Type eventType) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EntityEventBus.Broadcast.cs:line 324
   at Robust.Shared.GameObjects.EntityEventBus.RaiseEvent(EventSource source, Object toRaise) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EntityEventBus.Broadcast.cs:line 267
   at Robust.Server.GameObjects.ServerEntityManager.<Initialize>b__8_0(Object _, Object systemMsg) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\GameObjects\ServerEntityManager.cs:line 53
   at Robust.Server.GameObjects.ServerEntityManager.DispatchEntityNetworkMessage(MsgEntity message) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\GameObjects\ServerEntityManager.cs:line 266
   at Robust.Server.GameObjects.ServerEntityManager.TickUpdate(Single frameTime, Boolean noPredictions, Histogram histogram) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\GameObjects\ServerEntityManager.cs:line 192
   at Robust.Server.BaseServer.Update(FrameEventArgs frameEventArgs) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\BaseServer.cs:line 731
   at Robust.Server.BaseServer.<SetupMainLoop>b__67_1(Object sender, FrameEventArgs args) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\BaseServer.cs:line 544
   at Robust.Shared.Timing.GameLoop.Run() in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\Timing\GameLoop.cs:line 235
   at Robust.Server.BaseServer.MainLoop() in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\BaseServer.cs:line 571
   at Robust.Server.Program.ParsedMain(CommandLineArgs args, Boolean contentStart, ServerOptions options) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\Program.cs:line 74
   at Robust.Server.Program.Start(String[] args, ServerOptions options, Boolean contentStart) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\Program.cs:line 42
   at Robust.Server.ContentStart.Start(String[] args) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\ContentStart.cs:line 10
   at Content.Server.Program.Main(String[] args) in C:\Users\Sol\source\repos\space-station-14\Content.Server\Program.cs:line 9
Unhandled exception. Robust.Shared.Utility.DebugAssertException: Exception of type 'Robust.Shared.Utility.DebugAssertException' was thrown.
   at Robust.Shared.Utility.DebugTools.Assert(Boolean condition) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\Utility\DebugTools.cs:line 37
   at Content.Shared.Interaction.SharedInteractionSystem.InteractDoAfter(EntityUid user, EntityUid used, Nullable`1 target, EntityCoordinates clickLocation, Boolean canReach, Boolean checkDeletion) in C:\Users\Sol\source\repos\space-station-14\Content.Shared\Interaction\SharedInteractionSystem.cs:line 1065
   at Content.Shared.Interaction.SharedInteractionSystem.InteractUsing(EntityUid user, EntityUid used, EntityUid target, EntityCoordinates clickLocation, Boolean checkCanInteract, Boolean checkCanUse) in C:\Users\Sol\source\repos\space-station-14\Content.Shared\Interaction\SharedInteractionSystem.cs:line 1025
   at Content.Shared.Interaction.SharedInteractionSystem.UserInteraction(EntityUid user, EntityCoordinates coordinates, Nullable`1 target, Boolean altInteract, Boolean checkCanInteract, Boolean checkAccess, Boolean checkCanUse) in C:\Users\Sol\source\repos\space-station-14\Content.Shared\Interaction\SharedInteractionSystem.cs:line 440
   at Content.Shared.Interaction.SharedInteractionSystem.HandleUseInteraction(ICommonSession session, EntityCoordinates coords, EntityUid uid) in C:\Users\Sol\source\repos\space-station-14\Content.Shared\Interaction\SharedInteractionSystem.cs:line 314
   at Robust.Shared.Input.Binding.PointerInputCmdHandler.<>c__DisplayClass5_0.<.ctor>b__0(PointerInputCmdArgs& args) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\Input\Binding\InputCmdHandler.cs:line 97
   at Robust.Shared.Input.Binding.PointerInputCmdHandler.HandleCmdMessage(IEntityManager entManager, ICommonSession session, IFullInputCmdMessage message) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\Input\Binding\InputCmdHandler.cs:line 127
   at Robust.Server.GameObjects.InputSystem.InputMessageHandler(InputCmdMessage message, EntitySessionEventArgs eventArgs) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\GameObjects\EntitySystems\InputSystem.cs:line 65
   at Robust.Shared.GameObjects.EventBusExt.HandlerWrapper`1.Invoke(EntitySessionMessage`1 msg) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EventBusExt.cs:line 46
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass4_0`1.<SubscribeEvent>b__0(Unit& ev) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EntityEventBus.Broadcast.cs:line 173
   at Robust.Shared.GameObjects.EntityEventBus.ProcessSingleEventCore(EventSource source, Unit& unitRef, EventData subs) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EntityEventBus.Broadcast.cs:line 335
   at Robust.Shared.GameObjects.EntityEventBus.ProcessSingleEvent(EventSource source, Unit& unitRef, Type eventType) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EntityEventBus.Broadcast.cs:line 324
   at Robust.Shared.GameObjects.EntityEventBus.RaiseEvent(EventSource source, Object toRaise) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EntityEventBus.Broadcast.cs:line 267
   at Robust.Server.GameObjects.ServerEntityManager.<Initialize>b__8_0(Object _, Object systemMsg) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\GameObjects\ServerEntityManager.cs:line 53
   at Robust.Server.GameObjects.ServerEntityManager.DispatchEntityNetworkMessage(MsgEntity message) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\GameObjects\ServerEntityManager.cs:line 266
   at Robust.Server.GameObjects.ServerEntityManager.TickUpdate(Single frameTime, Boolean noPredictions, Histogram histogram) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\GameObjects\ServerEntityManager.cs:line 192
   at Robust.Server.BaseServer.Update(FrameEventArgs frameEventArgs) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\BaseServer.cs:line 731
   at Robust.Server.BaseServer.<SetupMainLoop>b__67_1(Object sender, FrameEventArgs args) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\BaseServer.cs:line 544
   at Robust.Shared.Timing.GameLoop.Run() in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Shared\Timing\GameLoop.cs:line 235
   at Robust.Server.BaseServer.MainLoop() in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\BaseServer.cs:line 571
   at Robust.Server.Program.ParsedMain(CommandLineArgs args, Boolean contentStart, ServerOptions options) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\Program.cs:line 74
   at Robust.Server.Program.Start(String[] args, ServerOptions options, Boolean contentStart) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\Program.cs:line 42
   at Robust.Server.ContentStart.Start(String[] args) in C:\Users\Sol\source\repos\space-station-14\RobustToolbox\Robust.Server\ContentStart.cs:line 10
   at Content.Server.Program.Main(String[] args) in C:\Users\Sol\source\repos\space-station-14\Content.Server\Program.cs:line 9
```

**Additional context**
The server is running the current build.

### PR DESCRIPTION
**Pull request by ewokswagger**
## About the PR
Fixed a bug in which putting Research disks into R&D Server crashes the game.

## Why / Balance
Easier to test with R&D servers now. Fixes #33188

## Technical details
added args.Handled = true; to OnAfterInteract in ResearchDiskSystem.

## Media
image

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
no

**Changelog**
no cl

### CODE REVIEW

File: Content.Server/Research/Disk/ResearchDiskSystem.cs [hunk 1]
```diff
@@ -23,14 +23,22 @@ public override void Initialize()
...
+            {
+                args.Handled = true;
                 return;
+            }
+
 
             _research.ModifyServerPoints(args.Target.Value, component.Points, server);
             _popupSystem.PopupEntity(Loc.GetString("research-disk-inserted", ("points", component.Points)), args.Target.Value, args.User);
             EntityManager.QueueDeleteEntity(uid);
+            args.Handled = true;
```
Comment (metalgearsloth) [inline Content.Server/Research/Disk/ResearchDiskSystem.cs:41]: 1. You're marking it as handled when it's not actually handled.
2. Setting it in all codepaths is the same as just setting it at the start of the method anyway.

You should remove the first two.

**Review by slarticodefast — APPROVED** [review]
Works as intended, thanks for the fix!
Could you remove the changelog? Since this is a debug assert it does not crash the game in the release configuration. So it has no impact players will notice.

**Comment by ewokswagger** [comment]
> Works as intended, thanks for the fix! Could you remove the changelog? Since this is a debug assert it does not crash the game in the release configuration. So it has no impact players will notice.

Alright, done.

**Comment by chromiumboy** [comment]
Thanks for the fix and your first contribution!

Tested in-game and works as intended

**Pull request merged**